client/mysqldump.c

```diff
@@ -2842,24 +2865,23 @@ static uint get_table_structure(char *table, char *db, char *table_type,
 
   if (!opt_xml && !mysql_query_with_error_report(mysql, 0, query_buff))
   {
-    /* using SHOW CREATE statement */
+    char buff[20+FN_REFLEN];
+    my_snprintf(buff, sizeof(buff), "show create table %s", result_table);
+
+    if (switch_character_set_results(mysql, "binary") ||
+        mysql_query_with_error_report(mysql, &result, buff) ||
+        switch_character_set_results(mysql, default_charset))
+    {
+      my_free(order_by);
+      order_by= 0;
+      DBUG_RETURN(0);
+    }
+
     if (!opt_no_create_info)
```

Comment:
please invert this if-else, to avoid double inversion

Reply:
What does double inversion mean?

Reply:
It turns out, that it's called [Reverse Conditional](https://www.refactoring.com/catalog/reverseConditional.html). Double negative is [here](https://refactoring.com/catalog/removeDoubleNegative.html).

TLDR `if (opt_no_create_info){...} else {...}` will be more readable

Reply:
No, in fact it won't. The main block which was there initially is more important. The second block is minor case.